src/Analyzer.JsonRuleEngine/Expressions/LeafExpression.cs

```diff
@@ -21,13 +21,14 @@ internal class LeafExpression : Expression
         /// <param name="jsonLineNumberResolver">An <see cref="ILineNumberResolver"/> to
         /// map JSON paths in evaluation results to the line number in the JSON evaluated.</param>
         /// <param name="operator">The operator used to evaluate the resource type and/or path.</param>
-        /// <param name="resourceType">The resource type this expression evaluates.</param>
-        /// <param name="path">The JSON path being evaluated.</param>
-        public LeafExpression(ILineNumberResolver jsonLineNumberResolver, LeafExpressionOperator @operator, string resourceType, string path)
-            : base(resourceType, path ?? throw new ArgumentNullException(nameof(path)))
+        /// <param name="commonProperties">The properties common across all <see cref="Expression"/> types.</param>
```

Comment:
These comments should likely list the properties so that you don't have to see the ExpressionCommonProperties class.

Reply:
I think the comments in `ExpressionCommonProperties` should be enough.  One benefit to having all these in a common class is so we don't have to keep on specifying what these are for every single expression - it's all localized in one source.  I can add a comment that `ExpressionCommonProperties.Path` is required here - unless we decide to make it optional (referenced in earlier comment in this PR).